### ISSUE TITLE
Fastspeech 2 phoneme 'Invalid argument:  Incompatible shapes' during training

### ISSUE THREAD

**Issue by PedroDKE** [TensorSpeech/TensorFlowTTS#512]
i'm trying to train a phoneme based FS2 model. I followed all the steps explained in the [libritts](https://github.com/TensorSpeech/TensorFlowTTS/tree/master/examples/fastspeech2_libritts) readme (i also did the fix mismatch step already). when I start to train my model, within the first 10 steps I get the following error:
```
(0) Invalid argument:  Incompatible shapes: [10,93,256] vs. [10,132,256]
	 [[node tf_fast_speech2/add_1 (defined at /home/user/anaconda3/envs/tts/lib/python3.8/site-packages/tensorflow_tts/models/fastspeech2.py:185) ]]
	 [[tf_fast_speech2/length_regulator/while/loop_body_control/_117/_111]]
(1) Invalid argument:  Incompatible shapes: [10,93,256] vs. [10,132,256]
	 [[node tf_fast_speech2/add_1 (defined at /home/user/anaconda3/envs/tts/lib/python3.8/site-packages/tensorflow_tts/models/fastspeech2.py:185) ]]
```

I already checked if all my duration,f0,energies,mfcc files have the same length and this does to be the case but yet something seems to be going wrong. From the error line i get that something is probably going wrong when summing the   last_encoder_hidden_states and the f0/energy gt embeddings during training and the sizes do not match here, but i'm unsure on why this is.

**Comment by dathudeptrai** [comment]
@PedroDKE you should check the len of (f0, energies) after reduce by its duration (https://github.com/TensorSpeech/TensorFlowTTS/blob/master/examples/fastspeech2_libritts/fastspeech2_dataset.py#L174-L175)

**Comment by PedroDKE** [comment]
@dathudeptrai thank you for the reply.
For all my files i checked if the length from the output from 'average_by_duration(norm_mean_std(f0/energy))' is equal to the length of my id's in the "utt_ids.npy" file and they are all be the same length.

**Comment by dathudeptrai** [comment]
@PedroDKE you should loop the dataset and check, you need to figure out what samples caused this problem, this issue is 100% caused by the dataset, not the model itself.

**Comment by PedroDKE** [comment]
okay so using the little script i found some files that caused the issue. By going a bit further with these files I figured that some of the validation files had their durations also in the training/fix_dur folder (not sure why, probably from an earlier preprocessing , and also not sure how this passed all the checks when asserting if the folder lengths are the same etc). So by removing my datafolder and doing the preprocessing all over again the issue got fixed. Thank you!
